### PR TITLE
Application interface for users

### PR DESCRIPTION
**Pull request by Tosiekdev**
One small question, should i call init and update in some sort of main loop?

### CODE REVIEW

File: include/Application.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+namespace arch {
+
+struct Application {
```
Comment (Chris-plusplus) [inline include/Application.h:5]: Polymorphic classes should use virtual destructor.

File: archimedes_bin/main.cpp [hunk 2]
```diff
@@ -1,22 +1,17 @@
...
-	arch::Logger::critical("Hello, {}!", "World");
-
-	arch::Logger::log(arch::LogLevel::info, "Info log");
-
 	arch::EngineConfig config {
 		600, 480, "Archimedes Test", glm::vec4(0, 0, 0, 0)
 	};
 
-	arch::Engine engine { config };
+	arch::Ref<MyApp> myApp = arch::createRef<MyApp>();
```
Comment (Chris-plusplus) [inline archimedes_bin/main.cpp:13]: I think myApp should be either moved to engine, or constructed in-place in engine's constructor, making it as reference is not necessary.

File: include/Engine.h [hunk 3]
```diff
@@ -20,7 +23,7 @@ struct EngineConfig {
  */
 class Engine {
 public:
-	Engine(const EngineConfig& config);
+	Engine(const EngineConfig& config, const Ref<Application>& application);
```
Comment (Chris-plusplus) [inline include/Engine.h:26]: as higher, move or construct in-place

File: include/Engine.h [hunk 4]
```diff
@@ -32,6 +35,7 @@ class Engine {
 	Window _window;
 	EngineConfig _engineConfig;
 	gfx::Renderer _renderer;
+	Ref<Application> _application;
```
Comment (Chris-plusplus) [inline include/Engine.h:38]: as higher, move or construct in-place

File: src/Engine.cpp [hunk 5]
```diff
@@ -9,7 +9,8 @@
 
 namespace arch {
 
-Engine::Engine(const EngineConfig& config): _window(1, 1, {}), _engineConfig(config) {}
+Engine::Engine(const EngineConfig& config, const Ref<Application>& application):
```
Comment (Chris-plusplus) [inline src/Engine.cpp:12]: as higher, move or construct in-place

**Review by Chris-plusplus — CHANGES_REQUESTED** [review]

File: archimedes_bin/main.cpp (hunk 2 above)
Comment (Twarug) [reply archimedes_bin/main.cpp:13]: We agreed upon this design on last meeting. It should be Ref.

**Review by Twarug — APPROVED** [review]

**Review by Chris-plusplus — APPROVED** [review]

**Pull request merged**